packages/superset-ui-number-format/src/formatters/D3Formatter.js

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable sort-keys */
+
+import isString from 'lodash/isString';
+import { format as d3Format } from 'd3-format';
+import { isRequired } from '@superset-ui/core';
+import NumberFormatter from '../NumberFormatter';
+
+export default class D3Formatter extends NumberFormatter {
+  constructor(configOrFormatString = isRequired('configOrFormatString')) {
+    const config = isString(configOrFormatString)
+      ? { formatName: configOrFormatString }
+      : configOrFormatString;
+
+    let formatFunc;
+    let isInvalid = false;
+
+    try {
+      formatFunc = d3Format(config.formatName);
+    } catch (e) {
+      formatFunc = () => `Invalid format: ${config.formatName}`;
+      isInvalid = true;
+    }
+
+    super({ ...config, formatFunc });
```

Comment:
should this be `{ formatFunc, ...config }` in case the user passed the function themselves? (unlikely I guess) Otherwise I don't fully understand the utility in supporting a config object + string.

Reply:
The order is correct. For `D3Formatter`, you either supply 

```js
// just the D3 format string
new D3Formatter('.2f');
// or accompany it with human-readable label and description
new D3Formatter({
  formatId: '.2f',
  label: 'Float with 2 decimal points',
  description: 'lorem ipsum dolor sit amet',
});
```

This will ensure that the `formatFunc` is native `d3-format`. If developer wants customization, just use plain `NumberFormatter`.

Reply:
I added this as comment in the code